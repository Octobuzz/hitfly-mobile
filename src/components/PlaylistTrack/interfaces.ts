export interface Playable {
  isPlaying?: boolean
}
