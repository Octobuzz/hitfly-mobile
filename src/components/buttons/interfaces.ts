export interface ButtonBase {
  title: string
  onPress?: () => void
  style?: any
}
