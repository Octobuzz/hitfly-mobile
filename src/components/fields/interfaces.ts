import { StyleProp, ViewStyle } from 'react-native'

export interface InputBase {
  onPress?: () => void
  onChangeText?: (text: string) => void
  onBlur?: (e: any) => void
  onFocus?: (e: any) => void
  value?: string
  defaultValue?: string
  editable?: boolean
  RightIcon?: JSX.Element
  label?: string
  error?: string
  style?: any
  containerStyle?: StyleProp<ViewStyle>
  forwardRef?: any
}

export interface CheckBoxProps {
  error?: string
  style?: StyleProp<ViewStyle>
}
