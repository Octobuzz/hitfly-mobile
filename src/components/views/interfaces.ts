export interface IView {
  noFill?: boolean
  noPadding?: boolean
  noVerticalPadding?: boolean
  noHorizontalPadding?: boolean
  paddingTop?: number
  paddingBottom?: number
  paddingVertical?: number
  paddingHorizontal?: number
  addBottomSafePadding?: boolean
  addTopSafePadding?: boolean
  addTopNavigationHeaderPadding?: boolean
}
