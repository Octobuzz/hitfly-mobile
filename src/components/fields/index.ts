export { default as Input } from './Input'
export { default as DatePicker } from './DatePicker'
export { default as CheckBoxUI } from './CheckBoxUI'
