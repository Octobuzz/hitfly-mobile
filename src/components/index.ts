export * from './views'
export * from './fields'
export * from './buttons'

export { default as TextBase } from './TextBase'
export { default as GenreItem } from './GenreItem'
export { default as TextWithLines } from './TextWithLines'
