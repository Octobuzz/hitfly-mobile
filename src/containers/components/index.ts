export * from './SocialAuth'

export { default as SocialAuth } from './SocialAuth'
export { default as HeaderRightButtons } from './HeaderRightButtons'
