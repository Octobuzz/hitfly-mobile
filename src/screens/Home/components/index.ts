export { default as StarsSection } from './StarsSection'
export { default as TracksSection } from './TracksSection'
export { default as SectionHeader } from './SectionHeader'
export { default as SectionWrapper } from './SectionWrapper'
export { default as PlaylistHeader } from './PlaylistHeader'
export { default as CollectionSection } from './CollectionSection'
