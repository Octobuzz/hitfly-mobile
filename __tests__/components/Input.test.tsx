import { render } from '../../jest/test-utils'
import React from 'react'
import { InputBase } from '../../src/components/fields/interfaces'
import { FieldProps } from 'formik'
import { Input } from 'src/components/fields'

describe('Input', () => {
  const props = {
    form: {
      errors: {},
      touched: {},
    },
    label: 'label',
    RightIcon: {},
    field: {
      name: 'name',
      onChange: (value: any) => value,
      onBlur: (value: any) => value,
      value: 123,
    },
  } as InputBase & FieldProps

  it('renders correctly', () => {
    const { asJSON } = render(<Input {...props} />)

    expect(asJSON()).toMatchSnapshot()
  })
})
