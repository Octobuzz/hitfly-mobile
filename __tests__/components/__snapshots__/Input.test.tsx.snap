// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    pointerEvents="auto"
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 56,
          "paddingBottom": 16,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "height": 4,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
            },
            Object {
              "left": 8,
              "right": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
                Object {
                  "translateX": 8,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.38)",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "bottom": -4,
                "left": -4,
                "position": "absolute",
                "right": -4,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "height": 4,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "width": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "rgba(0, 0, 0, 0.38)",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "bottom": -4,
              "left": -4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 4,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "rgba(0, 0, 0, 0.38)",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "bottom": 0.5,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": -4,
            }
          }
        />
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "height": 4,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": 4,
            },
            Object {
              "width": 8,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "borderColor": "rgba(0, 0, 0, 0.38)",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "bottom": -4,
              "left": 0,
              "position": "absolute",
              "right": -4,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "left": "-100%",
              "paddingLeft": "50%",
              "position": "absolute",
              "top": 0,
              "transform": Array [
                Object {
                  "scale": 0.75,
                },
                Object {
                  "translateY": -10,
                },
                Object {
                  "translateX": 0,
                },
              ],
              "width": "200%",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.38)",
                "fontFamily": "GothamPro",
                "fontSize": 16,
                "includeFontPadding": false,
                "lineHeight": 16,
                "textAlign": "left",
                "textAlignVertical": "top",
              }
            }
          >
            label
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="sentences"
            disableFullscreenUI={true}
            editable={true}
            rejectResponderTermination={true}
            selectionColor="#313131"
            style={
              Array [
                Object {
                  "flex": 1,
                  "includeFontPadding": false,
                  "margin": 0,
                  "padding": 0,
                  "paddingTop": 0,
                  "textAlign": "left",
                  "textAlignVertical": "top",
                  "top": 2,
                },
                Object {
                  "color": "#313131",
                  "fontSize": 16,
                  "height": 24,
                },
                Array [
                  Object {
                    "color": "#313131",
                    "fontFamily": "GothamPro",
                    "fontSize": 12,
                  },
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value="value"
          />
        </View>
      </View>
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "minHeight": 8,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          },
        ]
      }
    />
  </View>
</View>
`;
